js/main.js

```diff
@@ -4,19 +4,139 @@ var fullName = document.querySelector('#fullName');
 var $location = document.querySelector('#location');
 var bio = document.querySelector('#bio');
 var image = document.querySelector('.image');
-var button = document.querySelector('button');
+var container = document.querySelector('.container');
+var form = document.querySelector('form');
 
 avatar.addEventListener('input', function () {
   image.setAttribute('src', avatar.value);
 });
 
-button.addEventListener('click', function () {
+form.addEventListener('submit', function (event) {
+  event.preventDefault();
+  if (localStorage.getItem('data-object') === null) {
+    var data = {
+      view: 'edit-profile',
+      profile: {
+        username: '',
+        fullName: '',
+        location: '',
+        avatarUrl: '',
+        bio: ''
+      },
+      entries: []
+    };
+    data = JSON.stringify(data);
+    localStorage.setItem('data-object', data);
+  }
+
+  data = localStorage.getItem('data-object');
+  data = JSON.parse(data);
+
   data.profile.avatarUrl = avatar.value;
   data.profile.username = username.value;
   data.profile.fullName = fullName.value;
   data.profile.location = $location.value;
   data.profile.bio = bio.value;
   var dataVar = JSON.stringify(data);
   localStorage.setItem('data-object', dataVar);
-  document.reset();
+  viewSwapping('profile');
+  form.reset();
+});
+
+document.addEventListener('DOMContentLoaded', function () {
+  var data = localStorage.getItem('data-object');
+  data = JSON.parse(data);
+  if (data.profile.username !== '') {
+    viewSwapping('profile');
+  } else {
+    viewSwapping('edit-profile');
+  }
 });
+
+function render() {
+  var data = localStorage.getItem('data-object');
+  data = JSON.parse(data);
```

Comment:
You can access the data from the data object if you modifed it directly, you dont want to keep reaching into local storage